configure.ac

```
@@ -293,6 +293,7 @@ if test "$enable_man" = "yes"; then
 fi
 AM_CONDITIONAL(ENABLE_REGENERATE_MAN, test "x$enable_man" != "xno")
 
+AC_CHECK_HEADERS([sys/prctl.h], [], [enable_subids="no"])
```

Let says somebody wants to build shadow with suids ans explicitelly pass `--enable-subordinate-ids=yes` to configure.  But for some reason `sys/prctl.h` is missing. Configure check will silently pass but override the feature.
```
checking for DocBook XML DTD V4.5 in XML catalog... found
checking for DocBook XSL Stylesheets >= 1.70.1 in XML catalog... found
checking for sys/prctl.h... no
checking for sd_session_get_remote_host in -lsystemd... no
```

even though configure help says yes
```
sh$ ./configure --help | grep sub -A2
  --enable-subordinate-ids
                          support subordinate ids [default=yes]
  --enable-lastlog        enable lastlog [default=no]
  --enable-logind         enable logind [default=yes]
```

Instead of overriding default to `no`, we should provide fail with useful error message `AC_MSG_ERROR`.

We should inform user to either install missing `sys/prctl.h`  for building SUBUIDS( newgidmap newuidmap) or run configure `--disable-subordinate-ids`

---

Yes, let's make this obvious so that the project is set up correctly and explicitly.